From 11884fc73d2b252ae12ba4caa7b3a3f04256d82d Mon Sep 17 00:00:00 2001
From: Rushabh Shah <shahrs87@apache.org>
Date: Mon, 21 May 2018 12:33:00 -0500
Subject: [PATCH 2810/2848] Skip the proxy user check if the ugi has not been
 initialized. Contributed by Daryn Sharp

(cherry picked from commit 73e9120ad79c73703de21e0084591861813f3279)
(cherry picked from commit 04219e55c8983f88573b10205dbca5411e744b35)

CDH-68706

Change-Id: I93de45b47112a85149236fea202ee1b8250c8195
(cherry picked from commit 3e6f68264266ebd2d911375d62a3166deee808ae)
---
 .../java/org/apache/hadoop/conf/Configuration.java |    2 +-
 .../hadoop/security/UserGroupInformation.java      |    8 ++++++--
 2 files changed, 7 insertions(+), 3 deletions(-)

diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/conf/Configuration.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/conf/Configuration.java
index 92491d9..871f67c 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/conf/Configuration.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/conf/Configuration.java
@@ -237,7 +237,7 @@ public String toString() {
     }
 
     private static boolean getRestrictParserDefault(Object resource) {
-      if (resource instanceof String) {
+      if (resource instanceof String || !UserGroupInformation.isInitialized()) {
         return false;
       }
       UserGroupInformation user;
diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/UserGroupInformation.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/UserGroupInformation.java
index 4ca47eb..0350ea7 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/UserGroupInformation.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/security/UserGroupInformation.java
@@ -293,14 +293,18 @@ public boolean logout() throws LoginException {
   public static final String HADOOP_TOKEN_FILE_LOCATION = 
     "HADOOP_TOKEN_FILE_LOCATION";
   
+  public static boolean isInitialized() {
+    return conf != null;
+  }
+
   /** 
    * A method to initialize the fields that depend on a configuration.
    * Must be called before useKerberos or groups is used.
    */
   private static void ensureInitialized() {
-    if (conf == null) {
+    if (!isInitialized()) {
       synchronized(UserGroupInformation.class) {
-        if (conf == null) { // someone might have beat us
+        if (!isInitialized()) { // someone might have beat us
           initialize(new Configuration(), false);
         }
       }
-- 
1.7.9.5

